src/vm/costs/mod.rs

```diff
@@ -296,7 +599,7 @@ impl LimitedCostTracker {
         self.cost_function_references = m;
         self.cost_contracts = cost_contracts;
 
-        clarity_db.roll_back();
+        clarity_db.commit();
```

Comment:
Won't this cause a new MARF trie to be stored?

Reply:
Yeah, this is just creating and releasing `SAVEPOINT`s, it won't commit the trie (the `begin` and `commit` methods in the clarity.rs and clarity_db/marfed_kv modules are not very descriptively named).